rpc-ipc.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: page
+title: OpenTracing RPC/IPC Framework Implementation Guide
+---
+
+*Trace all the things!*
+
+<div id="toc"></div>
+
+## Audience
+
+The audience for this guide are developers interested in adding [OpenTracing](http://opentracing.io/) instrumentation to a web, RPC, or other framework that makes requests and/or receives responses. This instrumentation makes it easy for developers using the framework to incorporate end-to-end (distributed) tracing.
+
+Distributed tracing provides insight about individual requests as they propagate throughout a system. OpenTracing is an open-source standard API for consistent distributed tracing of requests across processes, from web and mobile client platforms to the storage systems and custom backends at the bottom of an application stack. Once OpenTracing integrates across  the entire application stack, it’s easy to trace requests across the distributed system. This allows developers and operators much-needed visibility optimize and stabilize production services.
+
+Before you begin, check [here](http://opentracing.io/integration/) to make sure that there's a working OpenTracing API for your platform.
+
+## Overview
+
+At a high level, here is what you need for an OpenTracing integration:
+
+Server framework requirements:
+
+* Filters, interceptors, middleware, or another way to process inbound requests
+* Active span storage: either a global context or a request-to-span map
+* Settings or another way to configure tracer configuration
+
+Client framework requirements:
+
+* Filters, interceptors, or another way to process outgoing requests
+* Settings or another way to configure tracing configuration
+
+## Pro-tips:
+
+Before we dive into implementation, here are a few important concepts and features that should be made available to framework users.
+
+### Operation Names
+
+You'll notice an operation_name variable floating around this tutorial. Every span is created with an operation name that should follow the guidelines outlined [here](http://opentracing.io/spec/#operation-names). You should have a default operation name for each span, but also provide a way for the user to specify custom operation names.
+
+Examples of default operation names:
+
+* The name of the request handler method
+* The name of a web resource
+* The concatenated names of an RPC service and method
+
+### Specifying Which Requests to Trace
+
+Some users may want to trace every request while other may want only specific requests to be traced. You should ideally allow users to set up tracing for either of these scenarios. For example, you could provide @Trace annotations/decorators, in which only annotated handler functions have tracing enabled. You can also provide settings for the user to specify whether they're using these annotations, versus whether they want all requests to be traced automatically.
+
+### Tracing Request Properties

```

Comment:
We should also suggest some sort of "Span Decorator"... E.g., this thing for GRPC-Go: https://github.com/grpc-ecosystem/grpc-opentracing/blob/master/go/otgrpc/options.go#L30

IMO it could actually obviate this "Tracing Request Properties" thing... let the user decide how to move framework state into the Span. It's not onerous because it's only done once.